packages/flutter_tools/test/flutter_manifest_test.dart

```diff
@@ -478,6 +481,28 @@ flutter:
         expectedBuildNumber: null,
       );
     });
+
+    testUsingContext('returns proper error when font detail is malformed', () async {
+      final BufferLogger logger = context.get<Logger>();
+      const String manifest = '''
+name: test
+dependencies:
+  flutter:
+    sdk: flutter
+flutter:
+  fonts:
+    - family: foo
+      fonts:
+        -asset: a/bar
```

Comment:
Should this be `- asset:`?

Reply:
This is the error case we're testing for from #31764